File: backend/app/schemas/assessment.py
from pydantic import BaseModel, Field
from typing import List

class AssessmentDetailRequest(BaseModel):
    risk_type_id: int
    inherent: float = Field(..., ge=1, le=5)
    kpmr: int = Field(..., ge=1, le=5)

class AssessmentSubmit(BaseModel):
    period_id: int
    unit_type: str # 'LPEI' atau 'UUS'
    details: List[AssessmentDetailRequest]

class AssessmentDetailResponse(BaseModel):
    risk_name: str
    weight_percent: str
    inherent_origin: float
    inherent_round: int
    kpmr: int
    risk_rating: int
    composite: float

class AssessmentResponse(BaseModel):
    id: int
    final_score: float
    final_rating: str
    table_data: List[AssessmentDetailResponse]

File: frontend/src/types/assessment.ts
export interface AssessmentDetailRequest {
  risk_type_id: number;
  inherent: number; // float 1-5
  kpmr: number; // integer 1-5
}

export interface AssessmentSubmit {
  period_id: number;
  unit_type: 'LPEI' | 'UUS';
  details: AssessmentDetailRequest[];
}

export interface AssessmentDetailResponse {
  risk_name: string;
  weight_percent: string;
  inherent_origin: number;
  inherent_round: number;
  kpmr: number;
  risk_rating: number;
  composite: number;
}

export interface AssessmentResponse {
  id: number;
  final_score: number;
  final_rating: string;
  table_data: AssessmentDetailResponse[];
}

export interface RiskType {
  id: number;
  name: string;
  description?: string;
}
